### PR TITLE
Code Insights: Unify lang stats insight fetching logic

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/31085

## Context

Prior to this PR we had two different gql handlers for fetching data for the lang stats insight (one for live preview on the creation UI, one on the dashboard page). In this PR we extract logic about lang stats insight from legacy CodeInsightsBackend class API to data fetching in one fetching hook (see this tracking issue to get more context on legacy class based API) https://github.com/sourcegraph/sourcegraph/issues/41145

## Test plan
- Check that lang stats card works as expected on the dashboard page
- Check lang stats creation UI live preview chart
- Check lang stats insight standalone insight page (to do it click on the lang stats insight card title on the dashboard page)

## App preview:

- [Web](https://sg-web-vk-lang-stats-insight-preview.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | -0.01% (-1.69 kb) | -0.01% (-1.69 kb) | 0.28% (+2) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vk/lang-stats-insight-preview/compare-report.html;tab=live_object) for a full comparison between the commits 9c451c5 and 4aaa2ba or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/enterprise/insights/core/hooks/live-preview-insight/use-live-preview-lang-stats-insight.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+interface LangStatsDatum {
+    name: string
+    totalLines: number
+    linkURL: string
+    fill: string
+}
+
+interface Props {
+    skip?: boolean
```
Comment (philipp-spiess) [inline client/web/src/enterprise/insights/core/hooks/live-preview-insight/use-live-preview-lang-stats-insight.ts:24]: I found `skip` a bit confusing. The difference, as far as I see, is that the query is not eagerly loaded but can still be lazily fetched. WDYT about making this more explicit by e.g. calling it `skipEagerLoading`?

File: client/web/src/enterprise/insights/core/hooks/live-preview-insight/use-live-preview-lang-stats-insight.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                .then(data => !hasRequestCanceled && setState({ status: LivePreviewStatus.Data, data }))
+                .catch(
+                    error => !hasRequestCanceled && setState({ status: LivePreviewStatus.Error, error: asError(error) })
+                )
+        } else {
+            setState({ status: LivePreviewStatus.Intact })
+        }
+
+        return () => {
+            hasRequestCanceled = true
```
Comment (philipp-spiess) [inline client/web/src/enterprise/insights/core/hooks/live-preview-insight/use-live-preview-lang-stats-insight.ts:65]: Nice!

**Review by philipp-spiess — APPROVED** [review]
Nice cleanup!

File: client/web/src/enterprise/insights/core/hooks/live-preview-insight/use-live-preview-lang-stats-insight.ts (hunk 1 above)
Comment (vovakulikov) [reply client/web/src/enterprise/insights/core/hooks/live-preview-insight/use-live-preview-lang-stats-insight.ts:24]: @philipp-spiess yeah I also was afraid that this wouldn't be so straightforward to understand from API perspective. 

This skip supposed to work as skip param in the Apollo use query. So when you're already in process of  fetching you can set skip: true and we stop the fetching and reset the internal state of the fetch hook. But also we have to pass it anyway (skip: true) if we want to work with lazyQuery. 

I think I should split these two scenarios into two different hooks (as Apollo does) like useLivePreview useLazyLivePreview. In this case skip would work only for the useLivePreview hook and we wouldn't have to pass skip:true with useLazyLivePreview.

**Pull request merged**